src/server/services/wfs/qgswfsgetfeature.cpp

```diff
@@ -1044,29 +1206,48 @@ namespace QgsWfs
     }
 
 
-    QDomElement createFeatureGML2( QgsFeature *feat, QDomDocument &doc, int prec, QgsCoordinateReferenceSystem &crs, const QgsAttributeList &attrIndexes, const QSet<QString> &excludedAttributes, const QString &typeName, bool withGeom, const QString &geometryName )
+    QDomElement createFeatureGML2( QgsFeature *feat, QDomDocument &doc, const createFeatureParams &params )
     {
       //gml:FeatureMember
       QDomElement featureElement = doc.createElement( QStringLiteral( "gml:featureMember" )/*wfs:FeatureMember*/ );
 
       //qgs:%TYPENAME%
-      QDomElement typeNameElement = doc.createElement( "qgs:" + typeName /*qgs:%TYPENAME%*/ );
-      typeNameElement.setAttribute( QStringLiteral( "fid" ), typeName + "." + QString::number( feat->id() ) );
+      QDomElement typeNameElement = doc.createElement( "qgs:" + params.typeName /*qgs:%TYPENAME%*/ );
+      typeNameElement.setAttribute( QStringLiteral( "fid" ), params.typeName + "." + QString::number( feat->id() ) );
       featureElement.appendChild( typeNameElement );
 
-      if ( withGeom && geometryName != QLatin1String( "NONE" ) )
+      if ( params.withGeom && params.geometryName != QLatin1String( "NONE" ) )
       {
         //add geometry column (as gml)
         QgsGeometry geom = feat->geometry();
 
+        int prec = params.precision;
+        QgsCoordinateReferenceSystem crs = params.crs;
+        QgsCoordinateTransform mTransform( crs, params.outputCrs );
+        try
+        {
+          QgsGeometry transformed = geom;
+          if ( transformed.transform( mTransform ) == 0 )
+          {
+            geom = transformed;
+            crs = params.outputCrs;
+            if ( crs.isGeographic() && !params.crs.isGeographic() )
+              prec = std::min( params.precision + 3, 6 );
```

Comment:
Is it standard to manage precision this way?

Reply:
In the ISO GeoJSON standard, the max precision  has been fixed to 6 for EPSG:4326, so I think I can homogenize the precision to 6 for all the Geographic requested outputCrs.

Reply:
Mm, after checking the standard, 6 digit is only a recommandation to avoid heavy files with 15 digits. 
As there is a per layer precision configuration windows, I would reuse the precision used for GML output. If really you need a different precision between GML an GeoJSON, we could add another column there, but I don't think it's worth it.
Also we could discuss if setting by default 6 digits there woudn't be a better behavior than the current digit default

Reply:
Sorry, last sentence was "the current 8 digit default"

Reply:
@haubourg the problem is not between GML or GeoJSON, is between meters and degrees. In WFS 1.1.0, the GML can be requested in an other CRS than the source. So if the features ares requested in a Geographic CRS, I can't used directly the precision, for exemple 1 for a CRS in meters. The test here is only for this type of transform.
The other point is that I have updated the current default precision to 6 digits to avoid heavy files.

Reply:
I can add the inverse, from Geographic to Planar, I can limit to 3 digits.